src/js/components/stock-movement-wizard/outboundImport/OutboundImport.jsx

```diff
@@ -70,6 +65,13 @@ const OutboundImport = () => {
     trigger,
   } = useOutboundImportForm({ next });
 
+  /** Redirect to first step if there is no cached data */
+  useEffect(() => {
+    if (_.isEmpty(cachedData) && !is(OutboundImportStep.DETAILS)) {
+      navigateToStep(OutboundImportStep.DETAILS);
+    }
+  }, []);
```

Comment:
let's move it to a separate small hook named e.g. useCachedData

Reply:
values from `localStorage` and `sessionStorage` are not related to state and will not trigger the component rerender even if you put the in the dependency array.
Also these values are synchronous unlike useEffect which is asynchronous which means that these values will be available in the hook without a delay.